mcp/motors/ec_motors.c

```diff
@@ -58,16 +58,40 @@ static ph_thread_t *ecmonitor_ctl_id;
 
 extern int16_t InCharge;
 
+#define ECAT_SENDRECV_STACKSIZE (64 * 1024)
+int ecat_sendrecv_cadence_ns = 2000000;
+pthread_t ecat_sendrecv_thread;
+pthread_t ecat_check_thread;
+// We require some globals here to keep track of state between threads for
+// EtherCAT setup and monitoring. Exercise care editing usage of these,
+// since we don't guard access to them.
+// Toggle for operation data to/from peripherals
+uint8_t ecat_do_sendrecv = 0;
+// The Working Counter is incremented if an EtherCAT device was successfully
+// addressed and a read operation, a write operation or a read/write operation
+// was executed successfully. We check it to verify that.
+uint16_t ecat_expected_wkc = 0;
+volatile uint16_t ecat_current_wkc = 0;
+// Keep track of whether all EtherCAT devices should be in the OP state. We use
+// this to monitor for devices that have departed the OP state.
+uint8_t ecat_in_op_state = 0;
+
+// Intentional aliasing
+int* p_ec_periphcount = &ec_slavecount;
+#define ec_periphcount (*p_ec_periphcount)
+ec_slavet* ec_periph = ec_slave;
+#define ec_periph (ec_periph)
+
```

Comment:
Just out of curiosity, if the goal is just having some aliases, why do we need `#define`s here instead of
```
int ec_periphcount = ec_slavecount;
int* ec_periph = ec_slave;
```

Reply:
To echo our slack conversation for posterity: these values are edited by SOEM routines at runtime, and I wanted a newly-named pointer to those values without adding dereferences all over our code. This is a preference of mine, a tradeoff between making most of the code people will read slightly more difficult vs. making these particular lines more difficult to read if someone needs to dig deeper occasionally.

Commit adds comments added to help guide the reader.